### PR TITLE
MM-69161 Adding permission check model to allow/deny requests (#678)

### PR DESCRIPTION
**Pull request by avasconcelos114**
#### Summary

This PR is a manual cherrypick of #678 onto release-1.12

## Change Impact: 🟡 Medium

**Reasoning:** This PR modifies webhook command authorization logic and error handling with behavioral changes to permission enforcement, but includes appropriate test coverage for the new scenarios. The shift from admin-only to namespace-based allowlisting is a deliberate design change with identified test cases covering the new behavior.

**Regression Risk:** Medium. The changes modify authorization checks (removal of admin-only requirement, addition of `isNamespaceAllowed()` validation), error handling (shift from string-based to typed error matching), and impact webhook command accessibility. Untested code paths may exist in edge cases around permission denial scenarios, though new tests cover the primary allowlist and forbidden flows.

**QA Recommendation:** Manual QA should focus on webhook command execution across different user roles and namespaces to verify the namespace allowlist behavior works as intended. Test the `webhook add` and `webhook list` commands with both allowed and disallowed namespaces. Verify error messages are user-friendly and appropriate. Since this is a cherrypick to a release branch, validate behavior against the main branch to ensure consistency. Manual QA is recommended due to the authorization behavior changes, though the presence of new test cases reduces overall risk.

### CODE REVIEW

**Review by nang2049 — APPROVED** [review]

File: server/audit.go [hunk 1]
```diff
@@ -64,6 +64,32 @@ func (p AttachCommentToIssueAuditResult) Auditable() map[string]any {
...
+type AddWebhookAuditResult struct {
+	HookID int    `json:"hook_id"`
+	URL    string `json:"url"`
+	Scope  string `json:"scope"`
+}
+
+func (p AddWebhookAuditResult) Auditable() map[string]any {
+	return map[string]any{"hook_id": p.HookID, "url": p.URL, "scope": p.Scope}
+}
+
```
[the remark this answers is not in the document]
Comment (avasconcelos114) [reply server/audit.go:92]: Issue noted and will resolve separately, but this is a cherry-pick being backported so it would be out of scope and place to address it here

File: server/command_test.go [hunk 2]
```diff
@@ -496,6 +521,7 @@ func TestAddWebhookCommand(t *testing.T) {
 			api := &plugintest.API{}
 			api.On("GetConfig", mock.Anything).Return(conf)
 			api.On("KVGet", "_usertoken").Return([]byte(encryptedToken), nil)
+			api.On("LogAuditRec", mock.Anything).Maybe()
```
[the remark this answers is not in the document]
Comment (avasconcelos114) [reply server/command_test.go:524]: Resolving this for the same reason as the other comment
